txp/Pos/Txp/Logic/Local.hs

```diff
@@ -123,10 +126,20 @@ txProcessTransactionNoLock itw@(txId, txAux) = reportTipMismatch $ runExceptT $
             { _ptcAdoptedBVData = bvd
             , _ptcUtxoBase = resolved
             }
-    pRes <-
-        lift $
-        modifyTxpLocalData $
-        processTxDo epoch ctx tipDB itw
+    pRes <- withTxpLocalData $ \txpData -> do
+        -- Note that we now have a (potentially) different UtxoModifier to the one
```

Comment:
Good question: I'm wondering whether we should simply avoid the double call to `getUtxoModifier` and move the `withTxpLocalData` block to incorporate also the call to `runDBToil`. I don't have a working knowledge of this part of the system to catch if this could cause a regression in behaviour, though.

Reply:
That was indeed my first attempt. But whilst `runDBToil` is just a coerce, `utxoGet` relies on some reader environment. So it's not currently possible to do this, without rewriting that code as well.

Reply:
Do you mean that `uv ≠ localUM`? If yes, then it's not a problem, because all this code is run when `StateLock` is taken, so nobody else can modify mempool.

Reply:
Ah, cool.